src/redisearch_rs/rlookup/src/load_document.rs

```diff
@@ -0,0 +1,390 @@
+/*
+ * Copyright (c) 2006-Present, Redis Ltd.
+ * All rights reserved.
+ *
+ * Licensed under your choice of the Redis Source Available License 2.0
+ * (RSALv2); or (b) the Server Side Public License v1 (SSPLv1); or (c) the
+ * GNU Affero General Public License v3 (AGPLv3).
+*/
+
+//! This module contains the functions that load documents into the lookup table.
+//! This functionality depends on the Redis Module API and the Redis JSON API.
+//!
+//! The function [`load_document`] is the main entry point for loading documents.
+
+use core::panic;
+use std::borrow::Cow;
+use std::ffi::CStr;
+use std::ptr::NonNull;
+
+use crate::RLookup;
+use crate::RLookupKey;
+use crate::RLookupKeyFlag;
+use crate::bindings::DocumentType;
+use crate::bindings::KeyMode;
+use crate::bindings::KeyTypes;
+use crate::bindings::RLookupCoerceType;
+use crate::bindings::RLookupLoadMode;
+use crate::bindings::RedisKey;
+use crate::bindings::RedisScanCursor;
+use crate::bindings::RedisString;
+use crate::bindings::ReplyTypes;
+use crate::bindings::call_hgetall;
+use crate::row::RLookupRow;
+use enumflags2::make_bitflags;
+use ffi::QueryError;
+use ffi::REDISMODULE_OK;
+use ffi::RSDocumentMetadata;
+use ffi::RedisSearchCtx;
+use sorting_vector::RSSortingVector;
+use value::RSValueFFI;
+
+/// Variables that are global in the C API and required for loading documents.
+///
+/// This is a temporary solution until we have a proper Rust port of the globals in the C API.
+pub struct GlobalOptions {
```

Comment:
Should we gather all the global variables we export like this to one place such that it will be easy to later remove this (once the non-temporary solution will be implemented)?